app/models/course.rb

```diff
@@ -1,6 +1,8 @@
 # frozen_string_literal: true
 
+# Course model
 class Course < ApplicationRecord
   belongs_to :schedule
   has_many :sections
+  has_many :instructor_preferences
```

Comment:
Make this dependent: :destroy.

Reply:
ok

Reply:
On that note...shouldn't all schedule dependents be destroyed on schedule destroy as well?
class Schedule < ApplicationRecord
  has_many :rooms
  has_many :instructors
  has_many :courses
  validates :schedule_name, :semester_name, presence: true
end

These ^^^

Reply:
Yes